drivers/hmis/app/graphql/mutations/create_enrollment.rb

```diff
@@ -0,0 +1,61 @@
+module Mutations
+  class CreateEnrollment < BaseMutation
+    argument :input, Types::HmisSchema::CreateEnrollmentValues, required: true
+
+    field :enrollments, [Types::HmisSchema::Enrollment], null: true
+    field :errors, [Types::HmisSchema::ErrorGroup], null: false
+
+    def resolve(input:)
+      user = hmis_user
+      errors = []
+
+      if input.household_members.select { |hm| hm.relationship_to_ho_h == 1 }.size != 1
+        return {
+          enrollments: [],
+          errors: [
+            {
+              id: 'input',
+              errors: [
+                ArgumentError.new('Exactly one client must be head of household'),
+              ],
+            },
+          ],
+        }
+      end
+
+      if Date.parse(input.start_date) > Date.today
+        return {
+          enrollments: [],
+          errors: [
+            {
+              id: 'input',
+              errors: [
+                ArgumentError.new('Entry date cannot be in the future'),
+              ],
+            },
+          ],
+        }
+      end
+
+      enrollments = input.to_enrollments_params.map do |attrs|
+        enrollment = Hmis::Hud::Enrollment.new(data_source_id: user.data_source_id, **attrs)
+
+        if enrollment.valid? && !enrollment.in_progress?
+          enrollment.save!
+        else
+          enrollment.in_progress = true
+          enrollment.save(validate: false)
+        end
```

Comment:
@eanders What do you think about that? The above is what we discussed, but I can see arguments either way. Maybe we should all discuss together?

Reply:
I could go either way.  This felt safer, the errors are still surfaced, but the data gets saved.

Reply:
Ok, as long as the validation errors are surfaced, that makes sense to me. I was thinking that the `save(validate: false)` would make it so that the validation errors don't get returned to the client.

Reply:
`validate: false` just means it'll save.  Calling `enrollment.valid?` above will generate the errors.

Reply:
ok! thank you

Reply:
Relatedly.. if the enrollment is saved as in progress then the Project should be nil. I'm not seeing that behavior when testing this out with graphiql. I tested by creating an in-progress enrollment and then looking it up in the console: `Hmis::Hud::Enrollment.find(58515).project` should be `nil`.